src/components/ArticleList.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from "react";
+import { fetchArticles } from "../api.js";
+
+const ArticleList = ({ articles, setArticles }) => {
+
+    useEffect(()=>{
+        fetchArticles(setArticles).then((articlesList) => {
+            setArticles(articlesList)
+        })
+    })
```

Comment:
This useEffect will need a dependency array, even if it's empty! Not including it can cause infinite re-renders 😱